src/state/mouseEventsMiddleware.ts

```diff
@@ -27,9 +28,11 @@ mouseMoveMiddleware.startListening({
   actionCreator: mouseMoveAction,
   effect: (action: PayloadAction<MousePointer>, listenerApi: ListenerEffectAPI<RechartsRootState, AppDispatch>) => {
     const mousePointer = action.payload;
-    const activeIndex: TooltipIndex = selectActiveIndexFromMousePointer(listenerApi.getState(), mousePointer);
-    if (activeIndex != null) {
-      listenerApi.dispatch(setMouseOverAxisIndex({ activeIndex, activeDataKey: undefined }));
+    const activeProps = selectActivePropsFromMousePointer(listenerApi.getState(), mousePointer);
+    if (activeProps?.activeIndex != null) {
+      listenerApi.dispatch(setMouseOverAxisIndex({ activeIndex: activeProps.activeIndex, activeDataKey: undefined }));
+    } else {
+      listenerApi.dispatch(mouseLeaveChart());
```

Comment:
this is needed to clear tooltip state when mouse moves out of the `inRange` function, but not out of the svg

Reply:
This should be a comment in code